source/widgets/UnixWindow.ooc

```diff
@@ -22,26 +22,51 @@ use ooc-draw
 use ooc-opengl
 use ooc-x11
 
-version((unix || apple) && !gpuOff) {
-UnixWindow: class extends DisplayWindow {
-	_xWindow: X11Window
-	_openGLWindow: OpenGLWindow
-	context ::= this _openGLWindow context
-	init: func (size: IntSize2D, title: String) {
-		super()
-		this _xWindow = X11Window new(size, title)
-		this _openGLWindow = OpenGLWindow new(this _xWindow)
+version(unix || apple) {
+	UnixWindowBase: class extends DisplayWindow {
+		_xWindow: X11Window
+		init: func (size: IntSize2D, title: String) {
+			super()
+			this _xWindow = X11Window new(size, title)
+		}
+		free: override func {
+			this _xWindow free()
+			super()
+		}
+		draw: override func (image: Image) {
+			if (image instanceOf?(RasterYuv420Semiplanar))
+				this _xWindow draw(image as RasterYuv420Semiplanar)
+			else {
+				raster := RasterYuv420Semiplanar convertFrom(image as RasterImage)
+				this _xWindow draw(raster)
+				raster referenceCount decrease()
+			}
+		}
 	}
-	free: override func {
-		this _openGLWindow free()
-		this _xWindow free()
-		super()
+version(gpuOff) {
```

Comment:
I think I would rather see the constructor for `UnixWindowBase` create and return one of two subclasses depending on the flag rather than having two separate classes inside version blocks. For example like you(?) did with the `Thread` class. I leave that to @davidhesselbom to decide, though.